### PR TITLE
add generic derivation for Address

### PR DESCRIPTION
**Pull request by benweitzman**
maybe some other types should have generic as well?

### CODE REVIEW

**Comment by akru** [comment]
Hi! This is a good idea!

So, it can be useful for types from:
- https://github.com/airalab/hs-web3/blob/master/src/Network/Ethereum/Web3/Types.hs
- https://github.com/airalab/hs-web3/blob/master/src/Network/Ethereum/Unit.hs

And also for generated by TH types from:
- https://github.com/airalab/hs-web3/blob/master/src/Network/Ethereum/Web3/TH.hs

**Comment by akru** [comment]
Hi! Do you want to add `Generic` derivations to listed types in this PR?

**Comment by benweitzman** [comment]
I added more generic instances, this is now a breaking change as it requires the `DeriveGeneric` language extension to be turned on when using TH functions. 

Also, a fix for encoding `bytesN` was committed to this repo, just wanted to make sure you saw that before merging!

File: web3.cabal [hunk 1]
```diff
@@ -52,6 +52,7 @@ library
                      , aeson
                      , text
   default-extensions:  OverloadedStrings
+                     , DeriveGeneric
```
Comment (akru) [inline web3.cabal]: I'm not sure that we should enable `DeriveGeneric` extension by default.

**Comment by aupiff** [comment]
What's the status of this PR? Adding Generic derivations to Address would help me. @benweitzman can we clean this up to be mergable and to not enable `DeriveGeneric` by default like @akru suggested?

**Pull request merged**